### ISSUE TITLE
[Bug]: querynode panic when searching and inserting

### ISSUE THREAD

**Issue by 10166** [milvus-io/milvus#21934]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: v2.2.2
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka):    pulsar
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.2.1
- OS(Ubuntu or CentOS): ubuntu
- CPU/Memory: 8c/32g
- GPU: null
- Others:
```

### Current Behavior

querynode panic when searching and inserting
The sdk throws some exceptions, such as:
![image](https://user-images.githubusercontent.com/13286541/216278917-faeb3640-1bc3-40ec-8bbb-6cc732f50d89.png)
![image](https://user-images.githubusercontent.com/13286541/216278944-add9ec37-c091-4c08-a5d6-a2a21459e631.png)

### Milvus Log

[Explore-logs-2023-02-02 15_29_48.txt](https://github.com/milvus-io/milvus/files/10566321/Explore-logs-2023-02-02.15_29_48.txt)
[logs.tar.gz](https://github.com/milvus-io/milvus/files/10567134/logs.tar.gz)

**Comment by xiaofan-luan** [comment]
there is no IndexNode available or etcd is not serviceable, wait a minute....
That's the only info I can see.
Could you collect full log with export-log.sh?

**Comment by xiaofan-luan** [comment]
see https://github.com/milvus-io/milvus/tree/master/deployments/export-log

**Comment by 10166** [comment]
[logs.tar.gz](https://github.com/milvus-io/milvus/files/10566840/logs.tar.gz)

**Comment by yanliang567** [comment]
does it indicate the growing segments make the querynode OOM?
```
libgomp: Thread creation failed: Resource temporarily unavailable
Thread creation failed: Resource temporarily unavailable
```

/assign @jiaoew1991 
/unassign

**Comment by 10166** [comment]
> does it indicate the growing segments make the querynode OOM?
> 
> ```
> libgomp: Thread creation failed: Resource temporarily unavailable
> Thread creation failed: Resource temporarily unavailable
> ```
> 
> /assign @jiaoew1991 /unassign

I guess this is not the reason, I checked the exit code of the pod is 1, oom should be 137

**Comment by jiaoew1991** [comment]
It is very likely that the `OpenMP` thread resources are exhausted when searching growing segments

**Comment by jiaoew1991** [comment]
/assign @liliu-z 
/unassign 
pls help this issue

**Comment by liliu-z** [comment]
Fixed in Knowhere 1.3.8/ Milvus 2.2.3
